aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -155,24 +155,49 @@ const KernelFunction& OperatorEntry::computeDispatchTableEntry(const c10::Dispat
 
 std::pair<const AnnotatedKernel&, const char*> OperatorEntry::computeDispatchTableEntryWithDebug(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) const {
   auto dispatch_ix = static_cast<uint8_t>(dispatch_key);
+  // dispatchTable contains entries for both runtime dispatch keys and alias dispatch keys.
+  // For any dispatch key, it'll pick a kernel using the following order:
+  //  (1) Use kernel if it's directly registered to this key
+  //  (2) Handle runtime keys that have kernels available from alias keys
+  //    (2.1) Use kernel from DispatchKey::Autograd if available
+  //    (2.2) Use catchAllKernel_(as if it was populated to DispatchKey::Autograd) if available
+  //  (3) Use fallthrough kernel that are registered as fallback.
+  //  (4) Use catchAll kernel if available
+  // XXX: currently Autograd is the only alias key, we'll update alias key precedence after we add new
+  //      alias keys AutogradDispatchCPUOrCUDA and Math.
+  // TODO: we can remove (2.2) and (4) after TypeDefault registrations are moved from catchAll to Math.
```

Comment:
ok good. 2.2 special case is pretty weird

Reply:
After trying out a few things I feel this is actually the most contained change to make it work for now and it can be easily removed once we have catchAll moved to Math. 
I'm thinking about keeping 2.2 in this PR and making removing catchAll as next task, wdyt?